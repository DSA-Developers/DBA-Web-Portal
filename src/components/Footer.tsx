import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import Nav from 'react-bootstrap/Nav';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';

import { FullWidth } from './Containers';

import logoFb from '../assets/facebook.svg';
import logoInst from '../assets/instagram.svg';
import logoGroup from '../assets/groupme.svg';
import DevLogo from '../assets/Dev_Logo.png';

const Footer: React.FC = ({}) => {
  return (
    <FullWidth className="bg-light border-top text-secondary" style={{ position: 'relative' }}>
      <StyledRow>
        <StyledCol>
          <Header>Site Map</Header>
          <Nav className="flex-column">
            <SiteLink as={Link} to="/home">
              Home
            </SiteLink>
            <SiteLink as={Link} to="/about">
              About
            </SiteLink>
            <SiteLink as={Link} to="/gallery">
              Gallery
            </SiteLink>
            <SiteLink as={Link} to="/contact">
              Contact
            </SiteLink>
          </Nav>
        </StyledCol>
        <StyledCol>
          <Header>Connect</Header>
          <SocialMedia>
            <a href="https://www.facebook.com/DSAatUF" target="_blank">
              <Logo src={logoFb}></Logo>
            </a>
            <a href="https://www.instagram.com/uf.dsa/" target="_blank">
              <Logo src={logoInst}></Logo>
            </a>
            <a href="https://groupme.com/join_group/53205471/hXrFzndj" target="_blank">
              <Logo src={logoGroup}></Logo>
            </a>
          </SocialMedia>
        </StyledCol>
        <StyledCol>
          <Header>Contact Us</Header>
          <p>Email: dsaatuf@gmail.com</p>
        </StyledCol>
      </StyledRow>
      <Credits>
        {/* <img src={DevLogo} style={{ width: '8%', height: 'auto' }} /> */}
        <DevelopedBy>
          <em>&ensp;Made with â™¥ by DSA Developers</em>
        </DevelopedBy>
      </Credits>
    </FullWidth>
  );
};

export default Footer;

const StyledRow = styled(Row)`
  display: flex;
  justify-content: center;
  height: fit-content;
  align-items: start;
  margin: 0;
  padding: 10vh 20% 10vh 20%;
`;

const StyledCol = styled(Col)`
  font-size: 1.5rem;
  flex-grow: 1;
  flex-shrink: 2;
  margin: 0 20px 0 20px;
  padding: 0;
  height: fit-content;
`;

const SocialMedia = styled.div`
  display: grid;
  grid-template-columns: repeat(3, 60px);
  gap: 10px;
`;

const Logo = styled.img`
  width: 50px;
  height: 50px;
  &:hover {
    transform: scale(1.2);
    transition-duration: 0.5s;
  }
`;

const Header = styled.h1`
  font-size: 2.5rem;
  margin: 0 0 0.5em 0;
  font-weight: 100;
  text-transform: uppercase;
`;

const SiteLink = styled(Nav.Link)`
  font-size: 1.5rem;
  font-weight: 100;
  color: rgba(0, 0, 0, 0.5);
  :hover {
    color: rgba(0, 0, 0, 0.7);
    text-decoration: none;
  }
  padding: 0 0 0.5em 0;
`;

const Credits = styled(Row)`
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  bottom: 0;
  width: 100%;
`;

const DevelopedBy = styled.p`
  position: relative;
  font-size: 25px;
  font-weight: 100;
  &:before {
    content: '';
    position: absolute;
    left: -60px;
    bottom: -20%;
    display: block;
    width: 60px;
    height: 55px;
    background-image: url(${DevLogo});
    background-size: contain;
  }
`;
