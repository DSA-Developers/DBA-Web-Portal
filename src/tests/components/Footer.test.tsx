import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Footer from '../../components/Footer';

describe('Navigation', () => {
  test('Footer renders', () => {
    const component = renderer.create(
      <Router>
        <Footer />
      </Router>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
