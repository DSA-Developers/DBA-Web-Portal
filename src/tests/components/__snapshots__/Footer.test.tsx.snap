// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Footer renders 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  bottom: 0;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c2 {
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  padding: 10vh 20% 10vh 20%;
}

.c3 {
  font-size: 1.5rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 2;
  -ms-flex-negative: 2;
  flex-shrink: 2;
  margin: 0 20px 0 20px;
  padding: 0;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(3,60px);
  gap: 20px;
}

.c7 {
  width: 60px;
  height: 60px;
}

.c4 {
  font-size: 2.5rem;
  margin: 0 0 0.5em 0;
  text-transform: uppercase;
}

.c5 {
  font-size: 1.5rem;
  color: rgba(0,0,0,0.5);
  padding: 0 0 0.5em 0;
}

.c5:hover {
  color: rgba(0,0,0,0.7);
}

<div
  className="c0 bg-light border-top text-secondary"
>
  <div
    className="c1 c2 row"
  >
    <div
      className="c3 col"
    >
      <h1
        className="c4"
      >
        Site Map
      </h1>
      <div
        className="flex-column nav"
        onKeyDown={[Function]}
      >
        <a
          className="c5"
          href="/home"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="c5"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          className="c5"
          href="/gallery"
          onClick={[Function]}
        >
          Gallery
        </a>
        <a
          className="c5"
          href="/contact"
          onClick={[Function]}
        >
          Contact
        </a>
      </div>
    </div>
    <div
      className="c3 col"
    >
      <h1
        className="c4"
      >
        Connect
      </h1>
      <div
        className="c6"
      >
        <a
          href="https://www.facebook.com/DSAatUF"
          target="_blank"
        >
          <img
            className="c7"
            src="facebook.svg"
          />
        </a>
        <a
          href="https://www.instagram.com/uf.dsa/"
          target="_blank"
        >
          <img
            className="c7"
            src="instagram.svg"
          />
        </a>
        <a
          href="https://groupme.com/join_group/53205471/hXrFzndj"
          target="_blank"
        >
          <img
            className="c7"
            src="groupme.svg"
          />
        </a>
      </div>
    </div>
    <div
      className="c3 col"
    >
      <h1
        className="c4"
      >
        Contact Us
      </h1>
      <p>
        <b>
          Email
        </b>
        : dsaatuf@gmail.com
      </p>
    </div>
  </div>
</div>
`;
